extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -1305,9 +1311,27 @@ export class AmpStoryPage extends AMP.BaseElement {
       return nextElement.id;
     }
 
+    // Link last page to first page if the story is on it's own.
```

Comment:
This `getNextPageId` refers to what's the next page in a regular navigation (tap next). Can you tell me more about why this fixes the bug? This `allowCircularReference ` + assuming that `swipe` capability is controlling whether we can replay the story is a bit hacky and I think we can find a better fix. 

My guess is that you want to set the right `distance` attribute?